tests/generative/generic_strategies.py

```diff
@@ -0,0 +1,40 @@
+from hypothesis import strategies as st
+
+
+# This is a variable that will normally be set to True, but which shrinking is
+# allowed to make False.
+#
+# Examples of where this is useful:
+#
+# * Turning off expensive test operations that you want to check but don't care
+#   about running if they turn out not to be relevant to the error you're
+#   seeing.
+# * Giving the shrinker a place to terminate generation in places where you've
+#   decided to do it yourself because e.g. the data was getting too large.
```

Comment:
Great, thank you! Feel free (though not obliged) to rebase this on to the original commit if you want. We tend to be quite liberal with rebasing (even mid-review, which I know is controversial) in order to prioritise the usefulness of the history.

Reply:
Noted! I'll rebase before merging.